src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3294,6 +3300,20 @@ export class ResultsViewPageStore {
         []
     );
 
+    readonly sampleSet = remoteData({
+        await: () => [this.samples],
+        invoke: () => {
+            return Promise.resolve(
+                ComplexKeySet.from(
+                    this.samples.result!.map(s => ({
```

Comment:
couldn't you just have used the unique id?

Reply:
the unique id is not stored in the comparison group object. just a list of study and sample ids